docs/DafnyRef/Statements.md

```diff
@@ -759,39 +797,43 @@ For example, `assert P; E` can be an expression. However, in
-It has a special type and that type indicates that the value is inspected to see if an abrupt return
-from the caller is warranted.
-This type is often a datatype, as shown in the examples above, but it may be any type with the appropriate members.
- * The restriction on the type of caller's first out-parameter is
-just that it must be possible (perhaps through generic instantiation and type inference, as in these examples) for `PropagateFailure` applied to the failure-compatible output from the callee to produce a value of the caller's first out-parameter type.
-If the caller's first out-parameter type is failure-compatible (which it need not be),
- then failures can be propagated up the call chain.
-If the keyword form of the statement is used, then no `PropagateFailure` member
-is needed and there is no restriction on the caller's first out-parameter.
- * In the statement `j, k :- Callee(i);`,
- when the callee's return value has an `Extract` member,
-the type of `j` is not the type of the first out-parameter of `Callee`.
-Rather it is a type assignable from the output type of `Extract` applied to the first out-value of `Callee`.
- * A method like `Callee` with a special first out-parameter type can still be used in the normal way:
-`r, k := Callee(i)`.
-Now `r` gets the first output value from `Callee`, of type `Status` or `Outcome<nat>` in the examples above.
-No special semantics or exceptional control paths apply.
-Subsequent code can do its own testing of the value of `r`
-and whatever other computations or control flow are desired.
- * The caller and callee can have any (positive) number of output arguments,
-as long as the callee's first out-parameter has a failure-compatible type
-and the caller's first out-parameter type matches `PropagateFailure`.
- * If there is more than one LHS, the LHSs must denote different l-values, unless the RHS is a list of expressions and the corresponding RHS values are equal.
- * The LHS l-values are evaluated before the RHS method call,
-in case the method call has side-effects or return values that modify the l-values prior to assignments being made.
+* The first out-parameter of the callee is special.
+  It has a special type and that type indicates that the value is inspected to see if an abrupt return
+  from the caller is warranted.
+  This type is often a datatype, as shown in the examples above, but it may be any type with the appropriate members.
+* The restriction on the type of caller's first out-parameter is
+  just that it must be possible (perhaps through generic instantiation and type inference, as in these examples) 
+  for `PropagateFailure` applied to the failure-compatible output from the callee to produce a value of 
+  the caller's first out-parameter type.
+  If the caller's first out-parameter type is failure-compatible (which it need not be),
+  then failures can be propagated up the call chain.
+  If the keyword form of the statement is used, then no `PropagateFailure` member
```

Comment:
Can you explicit ":- assume" for example?

Reply:
Done